pkg/cloud/aws/services/userdata/node.go

```diff
@@ -39,6 +39,30 @@ nodeRegistration:
 EOF
 
 kubeadm join --config /tmp/kubeadm-node.yaml
+`
+
+	nodeCloudInit = `{{.Header}}
+write_files:
+-   path: /tmp/kubeadm-node.yaml
+    owner: root:root
+    permissions: '0644'
+    content: |
+      ---
+      apiVersion: kubeadm.k8s.io/v1beta1
+      kind: JoinConfiguration
```

Comment:
Just to give some insight, I ran into some issues the last time I attempted this with the combined InitConfiguration and ClusterConfiguration for bootstrapping the initial control plane instance. For some reason at the time kubeadm didn't decode the combined config properly and trying to nest a second yaml document using `write_files` was giving me issues as well.

Reply:
I hit problems with `write_files` and certificates and keys. Solved it by introducing helper template function that converts certs and keys into base64 strings that `cloud-init` can recognise and handle correctly.